packages/firebase_remote_config/firebase_remote_config/lib/src/remote_config.dart

```diff
@@ -0,0 +1,161 @@
+// Copyright 2019 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+part of firebase_remote_config;
+
+/// The entry point for accessing Remote Config.
+///
+/// You can get an instance by calling [RemoteConfig.instance]. Note
+/// [RemoteConfig.instance] is async.
+class RemoteConfig extends FirebasePluginPlatform with ChangeNotifier {
+  // Cached instances of [FirebaseRemoteConfig].
+  static final Map<String, RemoteConfig> _firebaseRemoteConfigInstances = {};
+
+  // Cached and lazily loaded instance of [FirebaseRemoteConfigPlatform]
+  // to avoid creating a [MethodChannelFirebaseRemoteConfig] when not needed
+  // or creating an instance with the default app before a user specifies an
+  // app.
+  FirebaseRemoteConfigPlatform _delegatePackingProperty;
+
+  /// Returns the underlying delegate implementation.
+  ///
+  /// If called and no [_delegatePackingProperty] exists, it will first be
+  /// created and assigned before returning the delegate.
+  FirebaseRemoteConfigPlatform get _delegate {
+    if (_delegatePackingProperty == null) {
+      _delegatePackingProperty = FirebaseRemoteConfigPlatform.instanceFor(
+          app: app, pluginConstants: pluginConstants);
+    }
+    return _delegatePackingProperty;
+  }
+
+  /// The [FirebaseApp] this instance was initialized with.
+  final FirebaseApp app;
+
+  RemoteConfig._({this.app})
```

Comment:
Not sure what's the style guide used on this project is yet, but afaik Flutter places constructors first, then static methods, then properties

Reply:
@rrousselGit yeah this is something I wasn't sure on either. Do you know if there's something which can automate that on format? Keen to get something in place to make it consistent everywhere.

Reply:
There's a lint for constructors: `sort_constructors_first` 

I'm not aware of a lint for static members

Reply:
There is also: [`sort_unnamed_constructors_first`](https://dart-lang.github.io/linter/lints/sort_unnamed_constructors_first.html)

Reply:
Thanks, @kroikie don't worry about these for now, I'll get it applied across the whole project at some point.

Reply:
Sounds good.